mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/ResourceManagerGatewayCxn.java

```diff
@@ -114,10 +133,16 @@ protected Scheduler scheduler() {
 
     @Override
     public void startUp() throws Exception {
-        log.info("Trying to register with resource manager {}", gateway);
         try {
-            registerTaskExecutorWithRetry();
+            if (!alreadyRegistered.getState()) {
+                log.info("Trying to register with resource manager {}", gateway);
+                registerTaskExecutorWithRetry();
+            } else {
+                log.info("Registered with resource manager {} already", gateway);
+            }
+
             registered = true;
+            alreadyRegistered.setState(true);
         } catch (Exception e) {
```

Comment:
handle the IO error here? I guess we don't need to disconnect etc. if the alreadyRegistered file got corrupted.